tests/unit/s2n_session_cache_async_test.c

```diff
@@ -1,415 +0,0 @@
-/*
```

Comment:
why are we deleting this test?

Reply:
`s2n_session_cache_async_test` and `s2n_self_talk_session_id_test` are highly overlapped. The main difference is `cache_retrieve_callback` in `s2n_self_talk_session_id_test` is not blocking, and `cache_retrieve_callback` in `s2n_session_cache_async_test` would be blocked in the first call and unblocked for the second call. Thus, I don't think it is necessary to keep two tests separately. The test items in `s2n_session_cache_async_test` are merged into `s2n_self_talk_session_id_test`.